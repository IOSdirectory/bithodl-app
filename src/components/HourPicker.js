import React, { Component } from 'react';
import { StyleSheet, View, Picker } from 'react-native';
import PropTypes from 'prop-types';

const styles = StyleSheet.create({
  picker: {
    width: 100
  },
  borders: {
    position: 'absolute',
    top: 90.5,
    width: 100,
    height: 35,
    borderColor: '#CDCDCD',
    borderTopWidth: StyleSheet.hairlineWidth,
    borderBottomWidth: StyleSheet.hairlineWidth
  }
});

export default class HourPicker extends Component {
  _renderHourItems() {
    const hourItems = [];

    for (let hour = 0; hour < 24; hour++) {
      const label = hour < 10 ? `0${hour}:00` : `${hour}:00`;
      const value = hour.toString();

      hourItems.push(
        <Picker.Item key={hour} label={label} value={value} />
      );
    }

    return hourItems;
  }

  render() {
    return (
      <View>
        <Picker
          style={styles.picker}
          selectedValue={this.props.selectedValue}
          onValueChange={this.props.onValueChange}
        >
          {this._renderHourItems()}
        </Picker>
        <View style={styles.borders} pointerEvents='none' />
      </View>
    );
  }
}

HourPicker.propTypes = {
  selectedValue: PropTypes.string,
  onValueChange: PropTypes.func
};
