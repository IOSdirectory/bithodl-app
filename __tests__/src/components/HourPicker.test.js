import React from 'react';
import renderer from 'react-test-renderer';
import HourPicker from '../../../src/components/HourPicker';

describe('HourPicker', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <HourPicker />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
