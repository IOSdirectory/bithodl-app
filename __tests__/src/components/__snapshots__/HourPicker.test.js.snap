// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HourPicker renders correctly 1`] = `
<View>
  <Picker
    onValueChange={undefined}
    selectedValue={undefined}
    style={
      Object {
        "width": 100,
      }
    }
  >
    <Item
      label="00:00"
      value="0"
    />
    <Item
      label="01:00"
      value="1"
    />
    <Item
      label="02:00"
      value="2"
    />
    <Item
      label="03:00"
      value="3"
    />
    <Item
      label="04:00"
      value="4"
    />
    <Item
      label="05:00"
      value="5"
    />
    <Item
      label="06:00"
      value="6"
    />
    <Item
      label="07:00"
      value="7"
    />
    <Item
      label="08:00"
      value="8"
    />
    <Item
      label="09:00"
      value="9"
    />
    <Item
      label="10:00"
      value="10"
    />
    <Item
      label="11:00"
      value="11"
    />
    <Item
      label="12:00"
      value="12"
    />
    <Item
      label="13:00"
      value="13"
    />
    <Item
      label="14:00"
      value="14"
    />
    <Item
      label="15:00"
      value="15"
    />
    <Item
      label="16:00"
      value="16"
    />
    <Item
      label="17:00"
      value="17"
    />
    <Item
      label="18:00"
      value="18"
    />
    <Item
      label="19:00"
      value="19"
    />
    <Item
      label="20:00"
      value="20"
    />
    <Item
      label="21:00"
      value="21"
    />
    <Item
      label="22:00"
      value="22"
    />
    <Item
      label="23:00"
      value="23"
    />
  </Picker>
  <View
    pointerEvents="none"
    style={
      Object {
        "borderBottomWidth": 0.5,
        "borderColor": "#CDCDCD",
        "borderTopWidth": 0.5,
        "height": 35,
        "position": "absolute",
        "top": 90.5,
        "width": 100,
      }
    }
  />
</View>
`;
